bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/common/PckGenerator.java

```diff
@@ -777,4 +797,28 @@ private static int timeToRampValue(int timeMSec) {
         }
         return ret;
     }
+
+    /**
+     * Converts duration in milliseconds to lcntimer value
+     * Source: https://www.symcon.de/forum/threads/38603-LCN-Relais-Kurzzeit-Timer-umrechnen
+     *
+     * @param ms time in milliseconds
+     * @return lcn timer value
+     */
+    private static Integer convertMsecToLCNTimer(double ms) {
+        Integer lcntimer = -1;
+        if (ms >= 0 && ms <= 240960) {
+            double a = ms / 1000 / 0.03;
+            double b = (a / 32.0) + 1.0;
+            double c = Math.log(b) / Math.log(2);
+            double mod = Math.floor(c);
+            double faktor = 32 * (Math.pow(2, mod) - 1);
+            double offset = (a - faktor) / Math.pow(2, mod);
+            lcntimer = (new Double(offset + (mod * 32))).intValue();
```

Comment:
This should be sematically the same:
```suggestion
            lcntimer = (int) (offset + mod * 32);
```